src/mono/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.Mono.cs

```diff
@@ -1885,6 +1856,34 @@ private static bool IsValidGetMethodType(Type type)
             return false;
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
+            Justification = "Linker thinks Type.GetConstructor(ConstructorInfo) is one of the public APIs because it doesn't analyze method signatures. We already have ConstructorInfo.")]
+        public static ConstructorInfo GetConstructor(Type type, ConstructorInfo constructor)
+        {
+            if (!IsValidGetMethodType(type))
+                throw new ArgumentException("type is not TypeBuilder but " + type.GetType(), nameof(type));
+
+            if (type == null)
+                throw new ArgumentException("Type is not generic", nameof(type));
+
```

Comment:
I think this needs the same code that `GetMethod` has:

```csharp
            if (type is TypeBuilder && type.ContainsGenericParameters)
                type = type.MakeGenericType(type.GetGenericArguments());
```

(and same for GetField)

Reply:
I've tried, but it won't build.

Reply:
Looks like you also need
```
[UnconditionalSuppressMessage("ReflectionAnalysis", "IL2055:UnrecognizedReflectionPattern",
            Justification = "Type.MakeGenericType is used to create a typical instantiation")]
```